presto-main/src/main/java/com/facebook/presto/operator/LookupJoinPageBuilder.java

```diff
@@ -145,7 +145,9 @@ private void appendProbeIndex(JoinProbe probe)
         verify(previousPosition <= position);
         isSequentialProbeIndices &= position == previousPosition + 1 || previousPosition == -1;
 
+        // Update probe indices and size
```

Comment:
i think this comment doesn't explain much (given the code now does what it does).

btw, the cmt msg:
> in the extreme case where nothing is appended on the build
and same row appended on the probe can lead to the builder increasing
continuously.

I don't quite understand that.

Reply:
How about
 > Estimated size for probe should have both dictionary size and id size.
 Otherwise, in the extreme case where nothing is appended on the build
 during a cross join can lead to the builder appending probe indices
 indefinitely.

Check the unit test for more context.